src/api/app/assets/javascripts/webui2/package-view_file.js

```diff
@@ -1,31 +1,29 @@
-// FIXME: Refactor this code
 var DAYS = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
 var MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
 
+var toTwoChars = function(number, leadingChar) {
+  return (leadingChar + number).slice(-2);
+};
+
 // The jshint ignore:line is needed to ignore the offense: `'addChangesEntryTemplate' is defined but never used.` This is used in the view directly
 function addChangesEntryTemplate() { // jshint ignore:line
   var date = new Date(),
-      day = date.getUTCDate().toString(),
-      hours,
-      minutes,
-      seconds,
-      template;
-
-  if (day.length === 1) { day = " " + day; } // Pad single character day value
-  hours = date.getUTCHours();
-  if (hours < 10) { hours = '0' + hours; }
-  minutes = date.getUTCMinutes();
-  if (minutes < 10) { minutes = '0' + minutes; }
-  seconds = date.getUTCSeconds();
-  if (seconds < 10) { seconds = '0' + seconds; }
+      weekDay = DAYS[date.getUTCDay()],
+      month = MONTHS[date.getUTCMonth()],
+      day = toTwoChars(date.getUTCDate(), ' '),
+      hours = toTwoChars(date.getUTCHours(), '0'),
+      minutes = toTwoChars(date.getUTCMinutes(), '0'),
+      seconds = toTwoChars(date.getUTCSeconds(), '0'),
+      packagerName = $("a.changes-link").data('packagername'),
+      packagerEmail = $("a.changes-link").data('packageremail');
 
-  template = "-------------------------------------------------------------------\n" +
-    DAYS[date.getUTCDay()] + " " + MONTHS[date.getUTCMonth()] + " " + day + " " +
-    hours + ":" + minutes + ":" + seconds + " UTC " + date.getUTCFullYear() +
-    " - " + $("a.changes-link").data('packagername') +
-    " <" + $("a.changes-link").data('packageremail') + ">" +"\n\n" + "- \n" + "\n";
+  var template = "-------------------------------------------------------------------\n" +
+                 weekDay + " " + month + " " + day + " " +
+                 hours + ":" + minutes + ":" + seconds + " UTC " + date.getUTCFullYear() +
+                 " - " + packagerName + " <" + packagerEmail + ">\n\n- \n\n";
```

Comment:
[nitpick] Why don't you add the `editors[0].getValue()` here and name this variable as `newValue`??

Reply:
I think it is clear to have the templated separetely, it is already complicated to understand. You have to do another concatenation anyway. I could create an extra variable, so:

```javascript
newValue = template + editors[0].getValue();
editors[0].setValue(newValue);
```

@mdeniz would you prefer that option?

Reply:
No no.. is fine then